### DISCUSSION TITLE
Why twin.macro is so heavy

### DISCUSSION THREAD

**Discussion by itelo** [ben-rogerson/twin.macro#460] [Q&A] [answered]
First of all, awesome job in this library.

I'm planning to use `twin.macro` in my application, but when I went into [bundlephobia.com](https://bundlephobia.com/package/) to compare with other solutions to handle CSS I discovered that this library is much heavy than the others.

I will share some screenshot from [bundlephobia.com](https://bundlephobia.com/package/)

### twin.macro

<img width="300" alt="image" src="https://user-images.githubusercontent.com/11433064/122270282-86723980-ceb4-11eb-8fd8-e814927bc711.png">

### @emotion/react

<img width="300" alt="image" src="https://user-images.githubusercontent.com/11433064/122270539-cb966b80-ceb4-11eb-85aa-75cfd130edb2.png">

### @emotion/styled
<img width="300" alt="image" src="https://user-images.githubusercontent.com/11433064/122270574-d5b86a00-ceb4-11eb-8794-934d5067b691.png">

### styled-components

<img width="300" alt="image" src="https://user-images.githubusercontent.com/11433064/122270687-f54f9280-ceb4-11eb-8c82-1a708c2b3d48.png">

**Comment by ben-rogerson** [comment, marked as answer]
Thanks!

Because Twin is a babel plugin that runs at compile-time so it's not in the same category as these libraries.
The bundle size listed here never gets added to your app - only the resulting css rulesets get injected into your app.

The bundle size is largely due to the tailwind class definitions 👍
